app/src/main/java/app/wefridge/wefridge/model/Unit.kt

```diff
@@ -20,6 +21,17 @@ enum class Unit(val value: Int, @StringRes val _display: Int) {
 
         fun getByValue(value: Int?) = VALUES.firstOrNull { it.value == value }
 
+        fun getByString(symbol: String, callingFragment: Fragment): Unit? {
+            return when (symbol) {
+                callingFragment.getString(R.string.item_unit_gram) -> GRAM
+                callingFragment.getString(R.string.item_unit_kilogram) -> KILOGRAM
+                callingFragment.getString(R.string.item_unit_liter) -> LITER
+                callingFragment.getString(R.string.item_unit_milliliter) -> MILLILITER
+                callingFragment.getString(R.string.item_unit_ounce) -> OUNCE
+                callingFragment.getString(R.string.item_unit_piece) -> PIECE
```

Comment:
Possibly this can be simplified by finding an enum whose display value matches? Probably @PassiDel has a good idea here, so that only the enum contains mapping information, not the methods